plotter/labels_test.go

```diff
@@ -5,12 +5,114 @@
 package plotter_test
 
 import (
+	"image/color"
 	"testing"
 
+	"gonum.org/v1/plot"
 	"gonum.org/v1/plot/cmpimg"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg"
+	"gonum.org/v1/plot/vg/draw"
 )
 
 func TestLabels(t *testing.T) {
 	cmpimg.CheckPlot(ExampleLabels, t, "labels.png")
 	cmpimg.CheckPlot(ExampleLabels_inCanvasCoordinates, t, "labels_cnv_coords.png")
 }
+
+func TestLabelsWithGlyphBoxes(t *testing.T) {
+	cmpimg.CheckPlot(
+		func() {
+			const fontSize = 24
+
+			p, err := plot.New()
+			if err != nil {
+				t.Fatalf("could not create plot: %+v", err)
+			}
+
+			p.Title.Text = "Labels"
+			p.X.Min = -1
+			p.X.Max = +1
+			p.Y.Min = -1
+			p.Y.Max = +1
+
+			labels, err := plotter.NewLabels(plotter.XYLabels{
+				XYs: []plotter.XY{
+					{X: -0.8 + 0.00, Y: -0.5},
+					{X: -0.6 + 0.02, Y: -0.5},
+					{X: -0.4 + 0.04, Y: -0.5},
+					{X: -0.8 + 0.00, Y: +0.5},
+					{X: -0.6 + 0.02, Y: +0.5},
+					{X: -0.4 + 0.04, Y: +0.5},
+					{X: +0.0 + 0.00, Y: +0},
+					{X: +0.2 + 0.02, Y: +0},
+					{X: +0.4 + 0.04, Y: +0},
+				},
+				Labels: []string{
+					"Aq", "Aq", "Aq",
+					"Aq\nAq", "Aq\nAq", "Aq\nAq",
+
+					"Bg\nBg\nBg",
+					"Bg\nBg\nBg",
+					"Bg\nBg\nBg",
```

Comment:
The glyph boxes on 2-high and 3-high labels look tall. Do you know why this is? Is it expected?

Reply:
some amount of that headroom is probably for the descent of a possible line of text, above.
but I agree there is probably too much space.
I tried yesterday to play a bit with various things, to no avail.

there's probably some metric we are not taking into account.

![metrics](https://developer.apple.com/library/archive/documentation/TextFonts/Conceptual/CocoaTextArchitecture/Art/glyph_metrics_2x.png)

line gap perhaps isn't subtracted from the glyph box height?